src/blocksplitter.rs

```diff
@@ -157,11 +157,7 @@ pub fn blocksplit_lz77(options: &Options, lz77: &Lz77Store, maxblocks: usize, sp
     let mut lstart = 0;
     let mut lend = lz77.size();
 
-    loop {
-        if maxblocks > 0 && numblocks >= maxblocks {
-          break;
-        }
-
+    while maxblocks > 0 && numblocks >= maxblocks {
```

Comment:
This condition doesn't appear to be correct. Previously, the loop would *stop* when the condition was true, now it keeps going. The golden master files also have differences, but I'm not sure if it's from this line or not.